internal/provider/resource_database.go

```diff
@@ -338,6 +322,161 @@ func resourceDatabaseDelete(ctx context.Context, d *schema.ResourceData, meta in
+		if len(regionsToAdd) > 0 {
+			// add any regions to add first
+			if err := addRegionsToDatabase(ctx, d, client, regionsToAdd, databaseID, cloudProvider); err != nil {
+				return err
+			}
+		}
+		if len(regionsToDelete) > 0 {
+			// delete any regions that should be removed
+			if err := deleteRegionsFromDatabase(ctx, d, client, regionsToDelete, databaseID, cloudProvider); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func getRegionUpdates(oldRegions interface{}, newRegions interface{}) ([]string, []string){
+	mOld := map[string]bool{}
+	mNew := map[string]bool{}
+	var regionsToAdd []string
+	var regionsToDelete []string
+	// find any regions to add
+	for _, v := range oldRegions.([]interface{}) {
+		mOld[v.(string)] = true
+	}
+	for _, v := range newRegions.([]interface{}) {
+		mNew[v.(string)] = true
+	}
+	for _, v := range oldRegions.([]interface{}) {
+		if !mNew[v.(string)] {
+			regionsToDelete = append(regionsToDelete, v.(string))
+		}
+	}
+	for _, v := range newRegions.([]interface{}) {
+		if !mOld[v.(string)] {
+			regionsToAdd = append(regionsToAdd, v.(string))
+		}
+	}
+
+	return regionsToAdd, regionsToDelete
+}
+
+func addRegionsToDatabase(ctx context.Context, d *schema.ResourceData, client *astra.ClientWithResponses, regions []string, databaseID string, cloudProvider string) diag.Diagnostics {
+	// make sure the regions are valid
+	if err := ensureValidRegions(ctx, client, d); err != nil {
+		return err
+	}
+	// Currently, DevOps API only allows for adding 1 region at a time
+	for _, region := range regions {
+		datacenters := make([]astra.Datacenter, 1)
+		datacenters[0] = astra.Datacenter {
+			CloudProvider: cloudProvider,
+			Region:        region,
+			Tier:          "serverless",
+		}
+		resp, err := client.AddDatacentersWithResponse(ctx, astra.DatabaseIdParam(databaseID), datacenters)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		if resp.StatusCode() != http.StatusCreated {
+			return diag.FromErr(fmt.Errorf("Unexpected response addinng Regions: %s", string(resp.Body)))
+		}
+		// Wait for the database to be ACTIVE then set resource data
+		if err := waitForDatabaseAndUpdateResource(ctx, d, client, databaseID); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func deleteRegionsFromDatabase(ctx context.Context, d *schema.ResourceData, client *astra.ClientWithResponses, regions []string, databaseID string, cloudProvider string) diag.Diagnostics {
+	// get all the datacenetrs for the Datbase ID
+	dcListResp, err := client.ListDatacentersWithResponse(ctx, astra.DatabaseIdParam(databaseID), &astra.ListDatacentersParams{})
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if dcListResp.StatusCode() != http.StatusOK || dcListResp.JSON200 == nil {
+		return diag.FromErr(fmt.Errorf("Unexpected response fetching Datacenters: %s", dcListResp.Body))
+	}
+	dcs := *dcListResp.JSON200
+	// map DCs to region
+	m := map[string]astra.Datacenter{}
```

Comment:
Minor: could consider `m` to be more descriptive like `regionDCs` for readability.

Reply:
will do.